taichi/backends/opengl/shaders/random.glsl.h

```diff
@@ -5,12 +5,12 @@ STR(
 uvec4 _rand_;
 
 void _init_rand() {
-  uint i = (54321 + gl_GlobalInvocationID.x) * (12345 + _states_[0]);
+  uint i = (54321 + gl_GlobalInvocationID.x) * (12345 + _rand_state_);
   _rand_.x = 123456789 * i * 1000000007;
   _rand_.y = 362436069;
   _rand_.z = 521288629;
   _rand_.w = 88675123;
-  _states_[0] += 1;
+  _rand_state_ += 1;
```

Comment:
One quick question here: `_rand_state` is read and written in parallel (if I understand correctly). This is actually a data race. Is it intentional since you want its value to be *random*?

Reply:
Oh no! What I want is rand_state only increased once per-launch. Maybe we should do the increase work on CPU side. Thank for pointing out.
But currently this works well too, since we have GlobalInvocationId to make different seed, also.increase for n have no difference from 1. So it's still random in parallel kernel. (btw, do we have a test for this?) On the other side, reading and writing back to runtime buffer on CPU may harms performance, especially when kernel is small. So should we fix it, WDYT?

Reply:
Since it practically works and a perfect solution is hard, let's simply add the above discussions as comments there.